test/e2e/knative_eventing_test.go

```diff
@@ -0,0 +1,74 @@
+package e2e
+
+import (
+	"testing"
+
+	"github.com/openshift-knative/serverless-operator/test"
+	v1a1test "github.com/openshift-knative/serverless-operator/test/v1alpha1"
+)
+
+const (
+	knativeEventing = "knative-eventing"
+)
+
+var knativeControlPlaneDeploymentNames = []string{
+	"broker-controller",
+	"eventing-controller",
+	"eventing-webhook",
+	"imc-controller",
+	"imc-dispatcher",
+	"sources-controller",
+}
+
+func TestKnativeEventing(t *testing.T) {
+	caCtx := test.SetupClusterAdmin(t)
+
+	test.CleanupOnInterrupt(t, func() { test.CleanupAll(t, caCtx) })
+
+	t.Run("create subscription and wait for CSV to succeed", func(t *testing.T) {
+		if _, err := test.WithOperatorReady(caCtx, "serverless-operator-subscription"); err != nil {
+			t.Fatal("Failed", err)
+		}
+	})
+
+	t.Run("deploy knativeeventing cr and wait for it to be ready", func(t *testing.T) {
+		_, err := v1a1test.WithKnativeEventingReady(caCtx, knativeEventing, knativeEventing)
+		if err != nil {
+			t.Fatal("Failed to deploy KnativeEventing", err)
+		}
+	})
+
+	t.Run("verify correct deployment shape", func(t *testing.T) {
+		for i := range knativeControlPlaneDeploymentNames {
+			deploymentName := knativeControlPlaneDeploymentNames[i]
+			_, err := test.WithDeploymentReady(caCtx, deploymentName, knativeEventing)
+			if err != nil {
+				t.Fatalf("Deployment %s is not ready: %v", deploymentName, err)
+			}
+		}
+
+		err := test.WithDeploymentCount(caCtx, knativeEventing, len(knativeControlPlaneDeploymentNames))
+		if err != nil {
+			t.Fatalf("Deployment count in namespace %s is not the same as expected %d: %v", knativeEventing, len(knativeControlPlaneDeploymentNames), err)
+		}
+	})
+
+	t.Run("remove knativeeventing cr", func(t *testing.T) {
+		if err := v1a1test.DeleteKnativeEventing(caCtx, knativeEventing, knativeEventing); err != nil {
+			t.Fatal("Failed to remove Knative Eventing", err)
+		}
+
+		for i := range knativeControlPlaneDeploymentNames {
+			deploymentName := knativeControlPlaneDeploymentNames[i]
+			err := test.WithDeploymentGone(caCtx, deploymentName, knativeEventing)
+			if err != nil {
+				t.Fatalf("Deployment %s is not gone: %v", deploymentName, err)
+			}
+		}
+
+		err := test.WithDeploymentCount(caCtx, knativeEventing, 0)
+		if err != nil {
+			t.Fatalf("Some deployments were to be deleted but not in namespace %s. Err: %v", knativeEventing, err)
+		}
+	})
```

Comment:
This lacks a test for `"undeploy serverless operator and check dependent operators removed"` like we have in the serving suite. That should also fix the breaking CI.

Reply:
+1. perfect!

Reply:
We either add this test here or we remove the CSV/Subscription installation from each of the two suites. Or we make them gracefully accept things already installed.

Reply:
Just added that test